manifests/grubby.pp

```diff
@@ -0,0 +1,21 @@
+# Private class.
+class kernel::grubby{
+  if $caller_module_name != $module_name {
+    fail("Use of private class ${name} by ${caller_module_name}")
+  }
+
+  $grub_default_kernel = $kernel::grub_default_kernel ? {
+    'UNSET' => $kernel::kernel_version,
+    default => $kernel::grub_default_kernel,
+  }
+
+  # Under what circumstances could $grub_default_kernel *still*  == 'UNSET'?
+  if $kernel::set_default_kernel and $grub_default_kernel != 'UNSET' {
+    exec { 'set default kernel':
+      command => "/sbin/grubby --set-default=/boot/vmlinuz-${grub_default_kernel}",
+      path    => ['/bin','/usr/bin'],
+      unless  => "echo `/sbin/grubby --default-kernel` | grep -q ${grub_default_kernel}",
```

Comment:
Ok, I'll make those changes.  Sorry about the random use of echos and backticks!  No idea where my head could have been.